app/javascript/leaflet/controls/geosearch.js

```diff
@@ -0,0 +1,137 @@
+import { Control, setOptions } from "leaflet";
+import { debounce, boundsToBbox } from "../utils";
+
+export default class GeoSearchControl extends Control {
+  constructor(options) {
+    super(options);
+    setOptions(this, options);
+  }
+
+  onAdd(map) {
+    const container = document.createElement("div");
+    container.className = "leaflet-control search-control";
+    container.appendChild(this.options.staticButton);
+    container.appendChild(this.options.dynamicButton);
+
+    this._map = map;
+
+    const setButtons = () => {
+      if (document.getElementById("gbl-static-button")) {
+        this.staticButton = document.getElementById("gbl-static-button");
+        this.dynamicButton = document.getElementById("gbl-dynamic-button");
+        if (this.options.dynamic) {
+          this.staticButton.style.display = "none";
+          this.dynamicButton.style.display = "block";
+        } else {
+          this.staticButton.style.display = "block";
+          this.dynamicButton.style.display = "none";
+        }
+      }
+    };
+
+    setButtons();
+
+    const staticButtonExists = () => {
+      if (
+        typeof this.staticButton != "undefined" &&
+        this.staticButton != null
+      ) {
+        return true;
+      } else {
+        return null;
+      }
+    };
+
+    const staticSearcher = (e) => {
+      setButtons();
+      if (staticButtonExists() && e.target.id == "gbl-static-button") {
```

Comment:
Aren't lines 47-51 just doing what setButtons() is doing? I am not quite sure why this exists.

Reply:
Also this js is pretty broken. When you zoom it will refresh the page.